meinberlin/apps/plans/serializers.py

```diff
@@ -1,101 +1,111 @@
     district = serializers.SerializerMethodField()
-    point = serializers.SerializerMethodField()
+    organisation = serializers.SerializerMethodField()
     participation_active = serializers.SerializerMethodField()
     participation_string = serializers.SerializerMethodField()
+    point = serializers.SerializerMethodField()
     published_projects_count = serializers.SerializerMethodField()
-    organisation = serializers.SerializerMethodField()
-    created_or_modified = serializers.SerializerMethodField()
+    subtype = serializers.ReadOnlyField(default="plan")
     tile_image = serializers.SerializerMethodField()
     tile_image_alt_text = serializers.SerializerMethodField()
     tile_image_copyright = serializers.SerializerMethodField()
+    type = serializers.ReadOnlyField(default="plan")
+    url = serializers.SerializerMethodField()
 
     class Meta:
         model = Plan
         fields = [
-            "type",
-            "subtype",
-            "title",
-            "url",
-            "organisation",
-            "point",
-            "point_label",
             "cost",
+            "created_or_modified",
             "district",
-            "topics",
-            "status",
+            "organisation",
             "participation",
-            "participation_string",
             "participation_active",
+            "participation_string",
+            "point",
+            "point_label",
             "published_projects_count",
-            "created_or_modified",
+            "status",
+            "subtype",
             "tile_image",
             "tile_image_alt_text",
             "tile_image_copyright",
+            "title",
+            "topics",
+            "type",
+            "url",
         ]
 
-    def get_subtype(self, instance):
-        return "plan"
+    def get_url(self, instance: Plan) -> str:
+        return instance.get_absolute_url()
 
-    def _get_participation_status_plan(self, item):
-        return item.get_status_display(), not bool(item.status)
+    def get_published_projects_count(self, instance: Plan) -> int:
+        """
+        This method counts published projects of `instance`.
+        If the related field `instance.projects` has been prefetched (default of the plans view set, see `plans/api.py`)
+        then we count in Python, otherwise the database is hit with a `projects.count()`.
+        For details, see `docs/performance_of_plans_serializer.py`.
+        """
+        projects = instance.projects
+        projects_are_prefetched = False
 
-    def get_type(self, instance):
-        return "plan"
+        if projects:
+            try:
+                projects_are_prefetched = bool(
+                    projects.prefetch_cache_name in instance._prefetched_objects_cache
+                )
+            except AttributeError:
+                pass
 
-    def get_url(self, instance):
-        return instance.get_absolute_url()
+        if projects_are_prefetched:
+            count = 0
```

Comment:
maybe it wasn't clear, but we don't need the prefetch test, just do the count in python, and update the comment why we do so.

Reply:
ah yes, sry, done.